pandas/core/algorithms.py

```diff
@@ -216,7 +218,7 @@ def _ensure_data(values: ArrayLike) -> Tuple[np.ndarray, DtypeObj]:
 
 
 def _reconstruct_data(
-    values: ArrayLike, dtype: DtypeObj, original: AnyArrayLike
+    values: ArrayLike, dtype: Dtype, original: AnyArrayLike
```

Comment:
I think for internal low level functions we wanted to keep the dtype parameter restricted to dtype objects. The errors in this section were fixed in #40412.

I'll push a commit merging master.